### PR TITLE
HDS-711: New component for Highlight and Quote

### PR DESCRIPTION
**Pull request by minnas**
## Description

This PR adds component for Highlights and Quotes and needed documentation. More detailed description in the tickets below

## Motivation and Context

[HDS-711](https://helsinkisolutionoffice.atlassian.net/browse/HDS-711)
[HDS-712](https://helsinkisolutionoffice.atlassian.net/browse/HDS-712)
[HDS-1643](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1643)

## How Has This Been Tested?

Tested locally

## Demos

[React stories](https://city-of-helsinki.github.io/hds-demo/highlight-quote/?path=/story/components-highlight--default-highlight)
[Core stories](https://city-of-helsinki.github.io/hds-demo/highlight-quote-core/?path=/story/components-highlight--default-highlight)
[Documentation](https://city-of-helsinki.github.io/hds-demo/highlight-site/components/highlight)

[HDS-711]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-711?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HDS-712]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-712?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HDS-1643]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1643?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/components/highlight/tabs.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import { StatusLabel } from 'hds-react';
+
+import LeadParagraph from '../../../components/LeadParagraph';
+import Layout from '../../../components/layout';
+import PageTabs from '../../../components/PageTabs';
+
+# Highlight
+
+<StatusLabel type="alert" variant="rounded">Beta</StatusLabel>
```
Comment (harriplappalainen) [inline site/src/docs/components/highlight/tabs.mdx:14]: This is missing the Accessible StatusLabel and tooltip. Copy the implementation from another component's docs.

File: site/src/docs/components/highlight/tabs.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+<LeadParagraph>
+    Highlight component allows lifting an excerpt from a longer text for emphasis. Highlight supports two types: highlight (default) and quotation. The Quote element has a reference to the quote source. The Highlight element is an excerpt from the text without the source information.
+</LeadParagraph>
```
Comment (harriplappalainen) [inline site/src/docs/components/highlight/tabs.mdx:18]: To remove console warning about <p> inside a <p> tag, remove the new lines from the start and end of this paragraph. This kind of formatting is probably done by prettier but it causes said warnings.

File: site/src/docs/components/highlight/index.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  <Highlight type="quote" text="Add an interesting quote here" reference="First name Last name. Title."/>
+  <p>Sed mollis mauris nec dui condimentum pulvinar. Suspendisse vitae aliquet nisi. Donec libero ipsum, fermentum ut augue id, scelerisque lobortis orci.</p>
+</PlaygroundPreview>
+
+### Principles
+
+- Highlights are designed to capture users' attention. ** Therefore, they should be used sparingly. **
+- Use the prop `type="quote" ` to transform the highlight text into a quote with an optional reference, though we strongly recomend adding a reference to a quote
+- Pay attention to colours when using custom text or accent line color
+- Quote and Highlight adds vertical margin before and after actual content
```
Comment (harriplappalainen) [inline site/src/docs/components/highlight/index.mdx:32]: Dots at the end of bullet list items are missing.

File: site/src/docs/components/highlight/index.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  <p>Sed quis massa elementum, pulvinar lacus at, efficitur velit. Nam nec ornare augue. Aliquam pellentesque dictum nisl, a commodo arcu dapibus id.</p>
+  <Highlight type="quote" text="Add an interesting quote here" reference="First name Last name. Title."/>
+  <p>Sed mollis mauris nec dui condimentum pulvinar. Suspendisse vitae aliquet nisi. Donec libero ipsum, fermentum ut augue id, scelerisque lobortis orci.</p>
+</PlaygroundPreview>
+
+### Principles
+
+- Highlights are designed to capture users' attention. ** Therefore, they should be used sparingly. **
+- Use the prop `type="quote" ` to transform the highlight text into a quote with an optional reference, though we strongly recomend adding a reference to a quote
+- Pay attention to colours when using custom text or accent line color
```
Comment (harriplappalainen) [inline site/src/docs/components/highlight/index.mdx:31]: Pay attention how? 🤔 In regards to accessibility? If so then Accessibility tab might be a better place for this with a little more information here. Like "Pay attention that colours are accessible when --"

File: site/src/docs/components/highlight/code.mdx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+
+```jsx
+<>
+  <Highlight text="You may select an highlight from the article to be displayed here. Select an excerpt that you want the user to pay attention to." variant="l"/>
+  <Highlight text="You may select an highlight from the article to be displayed here. Select an excerpt that you want the user to pay attention to."/>
+  <Highlight text="You may select an highlight from the article to be displayed here. Select an excerpt that you want the user to pay attention to." variant="s"/>
+  <Highlight text="Add an interesting quote here" variant="l" type="quote" reference="First name Last name. Title."/>
+  <Highlight text="Add an interesting quote here" type="quote" reference="First name Last name. Title."/>
+  <Highlight text="Add an interesting quote here" variant="s" type="quote" reference="First name Last name. Title."/>
```
Comment (harriplappalainen) [inline site/src/docs/components/highlight/code.mdx:29]: The styles seem to be different for React and Core, why is that? 😮

**Review by harriplappalainen — COMMENTED** [review]
I thought the previous PR was done reviewing but I still found some things to improve perhaps 😅 But this is nearly there! Great effort so far 💪

File: site/src/docs/components/highlight/code.mdx (hunk 5 above)
Comment (minnas) [reply site/src/docs/components/highlight/code.mdx:29]: class for blockquote tag was missing from core  examples.

File: site/src/docs/components/highlight/tabs.mdx (hunk 1 above)
Comment (harriplappalainen) [reply site/src/docs/components/highlight/tabs.mdx:14]: The tooltip is missing too. If you look at for example `site/src/docs/components/accordion/tabs.mdx`, there's the StatuslabelTooltip component after these, and all of them are wrapped inside a `<div class="status-label-description">`.

**Review by harriplappalainen — APPROVED** [review]

**Pull request merged**